mesonpy/__init__.py

```diff
@@ -732,15 +732,39 @@ def _files_by_tag(self) -> Mapping[mesonpy._tags.Tag, Collection[str]]:
         files_by_tag: Dict[mesonpy._tags.Tag, List[str]] = collections.defaultdict(list)
         for file, details in self._install_plan.get('targets', {}).items():
             destination = pathlib.Path(details['destination'])
+
             # if in platlib, calculate the ABI tag
-            if (
-                not mesonpy._compat.is_relative_to(destination, '{py_platlib}')
-                and not mesonpy._compat.is_relative_to(destination, '{moduledir_shared}')
+            if not (
+                mesonpy._compat.is_relative_to(destination, '{py_platlib}')
+                or mesonpy._compat.is_relative_to(destination, '{moduledir_shared}')
             ):
-                continue
+                # XXX: Ideally we would just check the anchor placeholder in the
+                #      path (eg. check if it is {py_platlib}) but Meson seems to
+                #      have a bug where it sometimes returns a full path without
```

Comment:
Is there an upstream bug for this already? If not, maybe file one and update this comment?

Reply:
I did not open right away, but it exists now.

https://github.com/mesonbuild/meson/issues/10601